### DISCUSSION TITLE
FluentValidationAdapter should be updated

### DISCUSSION THREAD

**Discussion by sm-g** [canton7/Stylet#283] [Ideas]
https://github.com/canton7/Stylet/wiki/FluentValidationAdapter

with recent version of FluentValidation this code should be used:

```
public async Task<IEnumerable<string>> ValidatePropertyAsync(string propertyName)
        {
            // If someone's calling us synchronously, and ValidationAsync does not complete synchronously,
            // we'll deadlock unless we continue on another thread.
            return (await _validator.ValidateAsync(_subject, x => x.IncludeProperties(propertyName)).ConfigureAwait(false))
                .Errors.Select(x => x.ErrorMessage);
        }
```

**Comment by canton7** [comment]
Thanks, updated!

**Reply by MrCoffeeAtGitHub** [reply]
Oh, no. Now, I can't compile it since _subject is unknown. Just copy the implementation from the samples to the wiki.

public async Task<IEnumerable<string>> ValidatePropertyAsync(string propertyName)
        {
            // If someone's calling us synchronously, and ValidationAsync does not complete synchronously,
            // we'll deadlock unless we continue on another thread.
            return (await this.validator.ValidateAsync(this.subject, CancellationToken.None, propertyName).ConfigureAwait(false))
                .Errors.Select(x => x.ErrorMessage);
        }

P.S. The code-formatting didn't work properly :-(
